crates/topos-tce-broadcast/src/task_manager_futures/mod.rs

```diff
@@ -56,7 +59,9 @@ impl TaskManager {
         task_completion_sender: mpsc::Sender<(CertificateId, TaskStatus)>,
         subscription_view_receiver: mpsc::Receiver<SubscriptionsView>,
         event_sender: mpsc::Sender<ProtocolEvents>,
+        validator_id: ValidatorId,
         thresholds: ReliableBroadcastParams,
+        wallet: LocalWallet,
```

Comment:
`LocalWallet` will never change and doesn't need to be mutable. Put it in `Arc<LocalWallet>` to avoid clone